scripts/kind.sh

```diff
@@ -14,14 +14,34 @@ source "$SCRIPTS_DIR/lib/logging.sh"
 setup_kind_cluster() {
     local __cluster_name=$1
     local __controller_namespace=$2
-    local kubeconfig_path="$ROOT_DIR/build/clusters/$__cluster_name/kubeconfig"
+    local kind_cluster_build_dir="$ROOT_DIR/build/clusters/$__cluster_name"
+    local kubeconfig_path="$kind_cluster_build_dir/kubeconfig"
+    local kubecontext="kind-$__cluster_name"
+    local kubecontext_path="$kind_cluster_build_dir/kubecontext"
 
     info_msg "Creating cluster with name \"$__cluster_name\""
     _create_kind_cluster "$__cluster_name" "$kubeconfig_path"
 
-    info_msg "Exporting KUBECONFIG=$kubeconfig_path"
-    export KUBECONFIG=$kubeconfig_path
+    info_msg "Creating kubecontext file $kubecontext_path ... "
+    cat <<EOF > "$kubecontext_path"
+export KUBECONFIG="$kubeconfig_path"
+export KUBECONTEXT="$kubecontext"
+kubectl config use-context "$kubecontext"
+kubectl config set-context --current --namespace=ack-system
 
+controller_pod() {
+  echo \$(kubectl get pods -ojsonpath='{.items[0].metadata.name}')
```

Comment:
How do we know that `.items[0]` represents the controller under test?  Do we need to be concerned about a user's kind cluster running multiple ACK controllers?

Reply:
We don't :) This is just a best effort thing... that will work 99% of the time

Reply:
Should this be querying the `ack-system` namespace? Or are we installing these into `default`?

**Note: I was going to suggest the following, but I realise we don't actually have the controller type within the pod label**
Could we at least filter by label to ensure we are getting the right *type* of controller pod?
```
kubectl get pods -o=jsonpath='{.items[?(@.metadata.labels.name=="ack-s3-controller")].metadata.name}'
```

Reply:
> Should this be querying the `ack-system` namespace? Or are we installing these into `default`?

See line 30 above:

```diff
+ kubectl config set-context --current --namespace=ack-system
```

that sets the default namespace for the context to `ack-system` so ACK developers don't need to specify it.

> **Note: I was going to suggest the following, but I realise we don't actually have the controller type within the pod label** Could we at least filter by label to ensure we are getting the right _type_ of controller pod?
> 
> ```
> kubectl get pods -o=jsonpath='{.items[?(@.metadata.labels.name=="ack-s3-controller")].metadata.name}'
> ```

Yeah, I like the above idea...